crm_timesheet/models/crm_phonecall.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# License AGPL-3 - See LICENSE file on root folder for details
+##############################################################################
+
+from openerp import models, fields, api
+from datetime import datetime
+
+import logging
+from pprint import pformat
+_logger = logging.getLogger(__name__)
+
+
+class CrmPhonecall(models.Model):
+    _inherit = 'crm.phonecall'
+
+    analytic_account = fields.Many2one(
+        comodel_name='account.analytic.account')
+    timesheets = fields.One2many(comodel_name='hr.analytic.timesheet',
+                                 inverse_name='phonecall')
+
+    def _get_timesheet_vals(self, vals):
+        date = vals.get('date')
+        user_id = vals.get('user_id')
+        name = vals.get('name')
+        account_id = vals.get('analytic_account')
+        unit_amount = vals.get('duration')
+        res = {
+            'date': date[:10] if date else self.date[:10],
+            'user_id': user_id if user_id else self.user_id.id,
+            'name': name if name else self.name,
+            'account_id': account_id if account_id
+            else self.analytic_account.id,
+            'unit_amount': unit_amount / 60.0 if unit_amount
+            else self.duration / 60.0,
+            'code': 'phone',
+            'journal_id': self.env[
+                'hr.analytic.timesheet']._getAnalyticJournal()
+        }
+        _logger.info('TIMESHEET VALS : %s' % pformat(res))
+        return res
+
+    @api.model
+    def create(self, vals):
+        _logger.info('CREATE VALS : %s' % pformat(vals))
+        if vals.get('analytic_account') and vals.get('duration', 0) > 0:
+            timesheet_data = self._get_timesheet_vals(vals)
+            vals['timesheets'] = vals.get('timesheets') \
+                if vals.get('timesheets') else []
+            vals['timesheets'].append((0, 0, timesheet_data))
+        res = super(CrmPhonecall, self).create(vals)
+        return res
+
+    @api.one
+    def write(self, vals):
+        _logger.info('WRITE VALS : %s' % pformat(vals))
+        timesheet = self.env['hr.analytic.timesheet'].search([
```

Comment:
The one2many field exists. You need 2 fields for searching because you can add more timesheet activities in the form view. But only the timesheet activity automatically created will be updated.